File: Stack.py
# pilhas são estruturas de dados que podem adicionar dados ao final ou remover do final
class Stack:
    """Cria uma estrutura de dados stack(pilha)"""
    stack = []

    @classmethod
    def add(cls, valor):
        """Adiciona valor ao final da pilha"""
        Stack.stack.append(valor)

    @classmethod
    def pop(cls):
        """Remove e retorna o último valor"""
        valor = Stack.stack.pop()
        return valor

    @classmethod
    def peek(cls):
        """Retorna o último valor sem remover"""
        return Stack.stack[len(Stack.stack) - 1]

    @classmethod
    def size(cls):
        """retorna o  tamanho da pilha"""
        return len(Stack.stack)


class Main:
    @staticmethod
    def main():
        Stack.add(3)
        Stack.add("oi")
        print(Stack.stack)
        print(Stack.size())
        Stack.pop()
        print(Stack.stack)
        print(Stack.peek())


if __name__ == '__main__':
    Main.main()

File: Stack.java
// pilhas -> estruturas para armazenamento e remoção ao final
public class Stack {
     String[] vetor;
     int index , index1;
     //construtor
    Stack(int tamanho){
        vetor = new String[tamanho];
        index = 0;
        index1 = 0;
    }

    boolean isFull(){
        if(index1 == vetor.length) return false;
        else return true;
    }
     void addStack(String valor){
       if(isFull()){
           vetor[index1] = valor;
           index1 += 1;
       }else System.out.println("Stack isn't void!");
    }

    String removeStack(){
        String valor = vetor[index1 - 1];
        index1 -= 1;
        for(int i = 0; i < index1 - 1; i++){
            vetor[i] = vetor[i];
        }
        return valor;
    }
    void print(){
        System.out.print("[");
        for(int i = 0; i < index1; i ++){
            System.out.print(vetor[i] + ", ");
        }
        System.out.println("]");
    }

    public static void main(String[] args){
        Stack pilha = new Stack(10);
        pilha.addStack("4");
        pilha.addStack("2");
        pilha.addStack("4");
        pilha.addStack("5");
        pilha.print();
        System.out.println(pilha.removeStack());
    }
}
